src/Stan/Observation.hs

```diff
@@ -143,6 +118,41 @@ prettyShowObservation ReportSettings{..} Observation{..} = unlines $
         sols = inspectionSolution inspection
 
 
+prettyObservationSource
+    :: Bool  -- ^ Use colouring
+    -> Observation
+    -> [Text]
+prettyObservationSource isColour Observation{..} =
+    [ alignLine (n - 1)
+    , alignLine n <> getSourceLine
+    , alignLine (n + 1) <> arrows
+    ]
+  where
+    n :: Int
+    n = srcSpanStartLine observationLoc
+
+    alignLine :: Int -> Text
+    alignLine x = Text.justifyRight 4 ' ' (show x) <> " ┃ "
+
+    getSourceLine :: Text
+    getSourceLine = decodeUtf8 $
+        BS.lines observationFileContent !! (n - 1)
+
+    arrows :: Text
+    arrows = whenColour isColour (severityColour $ inspectionSeverity $ getInspectionById observationInspectionId)
+        <> Text.replicate start " "
+        <> Text.replicate arrow "^"
+        <> whenColour isColour reset
+      where
+        start = srcSpanStartCol observationLoc - 1
+        arrow = srcSpanEndCol observationLoc - start - 1
+
+{- | Checkes the predicate on colourfulness and returns an empty text when the
+colouroing is disabled.
+-}
+whenColour :: Bool -> Text -> Text
+whenColour isColour txt = if isColour then txt else ""
```

Comment:
Yeah, we really need to handle colouring in `colourista`... 😄

Reply:
Should we try the Implicit Parameters idea? :brain:

Reply:
@vrom911 I will look into the solution using the `reflection` library and will try to write a summary of this approach and a comparison between the two.